solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -185,6 +187,8 @@ public void prepare(ResponseBuilder rb) throws IOException
         if(rq instanceof RankQuery) {
           RankQuery rankQuery = (RankQuery)rq;
           rb.setRankQuery(rankQuery);
+          // we always need the score for reRanking
+          rb.setFieldFlags(rb.getFieldFlags() | GET_SCORES);
```

Comment:
https://github.com/apache/solr/pull/151/commits/909a25d59b826839dd34f4b1495b966913c95d3c proposes to continue to have only one `setFieldFlags` call

Reply:
That looks good to me! :)

One thing that I stumbled upon while looking at your change is, that we actually override the flags instead of just adding the score, because we start with this:
```java
int flags = 0;
```
A comment why we do this instead of starting with 
```java
int flags = rb.getFieldFlags();
```
would be great, I think.

Reply:
Great point. How about something along these lines?

```
int flags = 0; // we presume to be first to set this flag and hereby override any flags other components' prepare methods may have already set
```

Reply:
That's exactly what I was looking for, thank you. :)

Reply:
I appreciate all your ideas for improved comments! I included them in the commit where I added your comment above (dc2b84f)